assets/js/Script.js

```diff
@@ -0,0 +1,88 @@
+import { Carrito } from "./Carrito.js";
+import { Producto } from "./Producto.js";
+
+// viandas
+const productos = new Producto();
+const productosContainer = document.getElementById("productos");
+
+async function obtenerDatos() {
+  const API_URL = "https://67487dfa5801f51535912443.mockapi.io/Productos";
+  const spinner = document.getElementById("spinner");
+
+  spinner.style.display = "block";
+
+  return fetch(API_URL)
+    .then((response) => {
+      if (!response.ok) {
+        throw new Error("Error al obtener los datos");
+      }
+      return response.json();
+    })
+    .finally(() => {
+      spinner.style.display = "none";
+    });
+}
+
+document.addEventListener("DOMContentLoaded", () => {
+  obtenerDatos()
+    .then((datos) => productos.renderizarProductos(datos))
+    .then(() => {
+      const botones = document.querySelectorAll(".btn-enviar");
+      botones.forEach((boton) => {
+        boton.addEventListener("click", () => {
+          const producto = new Producto(
+            boton.dataset.id,
+            boton.dataset.foto,
+            boton.dataset.titulo,
+            boton.dataset.precio
+          );
+          carrito.agregar(producto);
+        });
+      });
+    })
+    .catch((error) => {
+      productosContainer.innerHTML = `
+        <li class="list-group-item text-danger">Error: ${error.message}</li>
+      `;
+    });
+});
+
+// carrito
+const carrito = new Carrito();
+
+carrito.actualizarCantidadAlCarritoVisual();
+carrito.renderizarProductosAlCarrito();
+
+// click a las viandas
+const botones = document.querySelectorAll(".btn-enviar");
```

Comment:
Evita la duplicación de código creando una función dedicada para agregar los event listeners a los botones. Esto hace que el código sea más fácil de mantener.
```suggestion
 function agregarEventListenerABotones() {
   const botones = document.querySelectorAll(".btn-enviar");
   botones.forEach((boton) => {
     boton.addEventListener("click", () => {
       const producto = new Producto(
         boton.dataset.id,
         boton.dataset.foto,
         boton.dataset.titulo,
         boton.dataset.precio
       );
       carrito.agregar(producto);
     });
   });
 }```